transform-message-formats/tests/test.bal

```diff
@@ -0,0 +1,203 @@
+import ballerina/http;
+import ballerina/test;
+
+final http:Client cl = check new (string `http://localhost:${port}/healthcare`);
+
+const GRAND_OAK_HOSPITAL = "grand oak community hospital";
+const CLEMENCY_MEDICAL_CENTER = "clemency medical center";
+const PINE_VALLEY_HOSPITAL = "pine valley community hospital";
+const THOMAS_COLLINS = "thomas collins";
+
+@test:Config
+function testSuccessfulReservation() returns error? {
+    ReservationResponse resp = check cl->/categories/surgery/reserve.post({
+        name: "John Doe",
+        dob: "1940-03-19",
+        ssn: "234-23-525",
+        address: "California",
+        phone: "8770586755",
+        email: "johndoe@gmail.com",
+        doctor: "thomas collins",
+        hospital_id: "grandoaks",
+        hospital: "grand oak community hospital",
+        card_no: "7844481124110331",
+        appointment_date: "2025-04-02"
+    });
+    ReservationResponse expResp = getSuccessfulAppointmentResponse(GRAND_OAK_HOSPITAL);
+    test:assertEquals(resp, expResp, "Response mismatched");
+}
+
+@test:Config
+function testInvalidHospital() {
+    ReservationResponse|http:ClientError resp = cl->/categories/surgery/reserve.post({
+        name: "John Doe",
+        dob: "1940-03-19",
+        ssn: "234-23-525",
+        address: "California",
+        phone: "8770586755",
+        email: "johndoe@gmail.com",
+        doctor: "thomas collins",
+        hospital_id: "monach",
+        hospital: "monarch hospital",
+        card_no: "7844481124110331",
+        appointment_date: "2025-04-02"
+    });
+
+    if resp !is http:ClientRequestError {
+        test:assertFail("expected an http:ClientRequestError, found " + (typeof resp).toString());
+    }
+
+    test:assertEquals(resp.message(), "Not Found");
+    var detail = resp.detail();
+    test:assertEquals(detail.statusCode, http:STATUS_NOT_FOUND);
+    test:assertEquals(detail.body, "Unknown hospital, doctor or category");
+}
+
+@test:Config
+function testInvalidDoctor() {
```

Comment:
Here, I have covered all the error-returning cases. What else we can have?

Reply:
I meant have one error test case instead of multiple. At the moment, we've duplicated a lot of the error tests across the projects, right? Just wondering if we are making it unnecessarily complicated.

Reply:
Ah. But I think this is fine for now. I'll keep this as a point for the next one.